<launch>
    <arg name="enable_ground_truth" default="true" />

    <arg name="distance_threshold" default="0.25"/>
    <arg name="yaw_threshold" default="0.2"/>
    <arg name="low_battery_secs" default="40"/>

    <group if="$(arg enable_ground_truth)">
        <node pkg="final_aerial_project" type="mission_planner" name="mission_planner" output="screen">
            <remap from="odom" to="/firefly/ground_truth/odometry"/>

            <param name="distance_threshold" value="$(arg distance_threshold)"/>
            <param name="yaw_threshold" value="$(arg yaw_threshold)"/>
            <param name="low_battery_secs" value="$(arg low_battery_secs)"/>
        </node>
    </group>

    <group unless="$(arg enable_ground_truth)">
        <node pkg="final_aerial_project" type="mission_planner" name="mission_planner" output="screen">
            <remap from="odom" to="/firefly/odom_filtered"/>
            <param name="distance_threshold" value="$(arg distance_threshold)"/>
            <param name="yaw_threshold" value="$(arg yaw_threshold)"/>
            <param name="low_battery_secs" value="$(arg low_battery_secs)"/>
        </node>
    </group>

</launch>
